anyway/infographics_utils.py

```diff
@@ -1191,6 +1187,89 @@ def pedestrian_injured_in_junctions_mock_data():  # Temporary for Frontend
         ]
 
 
+@register
+class AccidentTypeVehicleTypeRoadComparisonWidget(Widget):
+    name: str = "vehicle_accident_vs_all_accidents"  # WIP: change by vehicle type
+
+    def __init__(self, request_params: RequestParams):
+        super().__init__(request_params, type(self).name)
+        self.road_number = request_params.location_info["road1"]
+        # WIP: change rank, text by vehicle type
+        self.rank = 24
+        self.text = {"title": f"סוגי תאונות אופנועים בכביש {int(self.road_number)} בהשוואה לכל הארץ"}
+
+    def generate_items(self) -> None:
+        self.items = AccidentTypeVehicleTypeRoadComparisonWidget.accident_type_road_vs_all_count(
+            self.request_params.start_time, self.request_params.end_time, self.road_number
+        )
+
+    @staticmethod
+    def accident_type_road_vs_all_count(start_time, end_time, road_number):
+        location_all = "כל הארץ"
+        location_road = f"כביש {int(road_number)}"
+        logging.debug(f"start_time: {start_time} end_time: {end_time}")
+
+        num_accidents_label = "num_of_accidents"
+        vehicle_types = BE_CONST.MOTORCYCLE_VEHICLE_TYPES  # WIP: change by vehicle type
+        all_roads_query = (
+            db.session.query(AccidentMarker, Vehicle)
+            .with_entities(AccidentMarker.accident_type,
+                           func.count(distinct(AccidentMarker.provider_and_id)).label(num_accidents_label))
+            .join(Vehicle, Vehicle.provider_and_id == AccidentMarker.provider_and_id)
+            .filter(AccidentMarker.created >= start_time)
+            .filter(AccidentMarker.created < end_time)
+            .filter(Vehicle.vehicle_type.in_(vehicle_types))
+            .group_by(AccidentMarker.accident_type)
+            .order_by(desc(num_accidents_label))
+        )
+        all_roads_query_result = pd.read_sql_query(all_roads_query.statement, all_roads_query.session.bind)\
+            .to_dict(orient="records")  # pylint: disable=no-member
+        all_roads_sum_accidents = 0
+        all_roads_map = {}
+        for record in all_roads_query_result:
+            all_roads_sum_accidents += record[num_accidents_label]
+            all_roads_map[record[AccidentMarker.accident_type.name]] = record[num_accidents_label]
+
+        road_query = (
+            db.session.query(AccidentMarker, Vehicle)
+            .with_entities(AccidentMarker.accident_type,
+                           func.count(distinct(AccidentMarker.provider_and_id)).label(num_accidents_label))
+            .join(Vehicle, Vehicle.provider_and_id == AccidentMarker.provider_and_id)
+            .filter((AccidentMarker.road1 == road_number) | (AccidentMarker.road2 == road_number))
```

Comment:
The original query have `AND` and not an `OR` , so it needs to be:
`.filter((AccidentMarker.road1 == road_number) & (AccidentMarker.road2 == road_number))`

Reply:
AND doesn't make sense to me, because for a certain accident, a certain road number can be either in road1 or road2, nor both